File: infra-script/src/main/java/io/infra/structure/script/properties/ScriptProperties.java
package io.infra.structure.script.properties;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

/**
 * @author liuqinglin
 * Date: 2025/5/31 17:43
 */
@Data
@Configuration
@ConfigurationProperties(prefix = "infra.script")
public class ScriptProperties {
    /**
     * 是否启用脚本功能
     */
    private boolean enabled = true;
}

File: infra-script/src/main/kotlin/io/infra/structure/script/properties/ScriptProperties.kt
package io.infra.structure.script.properties

import io.infra.structure.script.constants.configPrefix
import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = configPrefix)
class ScriptProperties {
    /**
     * 是否启用脚本功能
     */
    var enabled: Boolean = true
}
